astroquery/heasarc/core.py

```diff
@@ -492,8 +492,14 @@ def locate_data(self, query_result=None, catalog_name=None):
             else:
                 query_result = self._last_result
 
+        if isinstance(query_result, Row):
+            query_result = query_result.table[[query_result.index]]
+
         if not isinstance(query_result, Table):
```

Comment:
I think this can be an elif now
```suggestion
        elif not isinstance(query_result, Table):
```